queue_services/entity-emailer/src/entity_emailer/message_tracker/tracker.py

```diff
@@ -61,6 +62,14 @@ def get_message_context_properties(queue_msg: nats.aio.client.Msg):
             identifier = email_msg.get('identifier', None)
             message_id = f'{etype}_{filing_id}'
             return create_message_context_properties(etype, message_id, None, identifier, False)
+        if etype == 'bc.registry.dissolution':
+            identifier = email_msg.get('identifier', None)
+            furnishing_id = email_msg['data']['furnishing']['furnishingId']
+            business = ar_overdue_stage_1_notification.get_business_by_furnishing_id(furnishing_id)
+            if business.legal_type in ['BC', 'ULC', 'CC', 'BEN']:
+                source = email_msg.get('source', None)
+                return create_message_context_properties(etype, message_id, source, identifier, False)
+            return create_message_context_properties(etype, None, None, None, False)
```

Comment:
This is not required as we are not sending the queue message to the emailer from the filer.  

When it's sent from the filer, we are using the already existing format which does not provide a unique "id" that can be used.  This why everything except `etype` == "bc.registry.names.request" is constructing a unique id based off of the incoming properties of a queue message.

We can just extend the check above for `if etype == 'bc.registry.names.request':` to include a check for `bc.registry.dissolution`

Reply:
Updated